src/libs/actions/Report.js

```diff
@@ -432,6 +432,67 @@ function unsubscribeFromReportChannel(reportID) {
     Pusher.unsubscribe(pusherChannelName);
 }
 
+/**
+ * Get the report ID, and then the actions, for a chat report for a specific
+ * set of participants
+ *
+ * @param {String[]} participants
+ */
+function fetchOrCreateChatReport(participants) {
+    let reportID;
+
+    if (participants.length < 2) {
+        throw new Error('fetchOrCreateChatReport() must have at least two participants');
+    }
+
+    API.CreateChatReport({
+        emailList: participants.join(','),
+    })
+
+        .then((data) => {
+            if (data.jsonCode !== 200) {
+                alert(data.message);
+                return;
+            }
+
+            // Set aside the reportID in a local variable so it can be accessed in the rest of the chain
+            reportID = data.reportID;
+
+            // Make a request to get all the information about the report
+            return API.Get({
+                returnValueList: 'reportStuff',
+                reportIDList: reportID,
+                shouldLoadOptionalKeys: true,
+            });
+        })
+
+        // Put the report object into Onyx
+        .then((data) => {
+            if (data.reports.length === 0) {
+                return;
+            }
+            const report = data.reports[reportID];
+
+            // Store only the absolute bare minimum of data in Onyx because space is limited
+            const newReport = getSimplifiedReportObject(report);
+            newReport.reportName = getChatReportName(report.sharedReportList);
+
+            // Optimistically update the last visited timestamp such that if the user immediately switches to another
+            // report the last visited order is still maintained.
+            newReport.lastVisitedTimestamp = Date.now();
+
+            // Merge the data into Onyx. Don't use set() here or multiSet() because then that would
+            // overwrite any existing data (like if they have unread messages)
+            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, newReport);
+
+            // Updates the personal details since its possible that a new participant was provided
+            PersonalDetails.getFromReportParticipants([newReport]);
+
+            // Redirect the logged in person to the new report
+            redirect(ROUTES.getReportRoute(reportID));
+        });
```

Comment:
Most of this is already handled by existing code. I believe the way it works is:

1. Once you have the `reportID`, then you put that into Onyx with 
```
Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`,{reportID});
```
2. There is code in this file `handleReportChanged()` which is listening for any new reports being put into Onyx 
3. If a new report doesn't have a report name (which yours doesn't, because you just put in the ID), then it will call `fetchChatReportsByIDs()`
4. This calls the API for `reportStuff` and adds the full information into Onyx automatically

So I think all you need to do is merge the reportID into Onyx with 
```
Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`,{reportID});
```
Then, redirect to the route like is there already, and the rest is automatic.

Does that make sense?

Reply:
Nice, this simplifies things a lot.